### PR TITLE
Added client and user portal e2e testing

### PR DESCRIPTION
**Pull request by NJavokhir**
### SECTION 1: RESOURCE SERVER (http://localhost:5001/)
- [x] http://localhost:5001/ home page loads
- [x] "LoginAsUser" button takes you to the user login page
- [x] Log in using "tester" and "12341234" should succeed
- [x] Updating the display name should work
- [x] Log out and Register a new user and try to log in with that new user
- [x] "LoginAsClient" button takes you to the client login page
- [x] Log in using "layer8" and "12341234" should work
- [x] You will be able to see your UID and Secret in profile
- [x] Log out and Register a new client and log in using that should work

### SECTION 2: WE'VE GOT POEMS (http://localhost:5173/)
- [x] Log in with 'tester' / '1234'
- [x] Log in anonymously with Layer8
- [x] Clicking "Get Next Poem" loads different poem correctly x 3
- [x] Clicking "Logout" takes you to the login screen
- [x] Clicking "Register" takes you to the registration page
- [x] Registering with a username, password, and profile image is successful
- [x] Logging in with the new username / password succeeds
- [x] Logging in with Layer8 opens the pop-up
- [x] Logging in with the newly registered credentials from Section 1 succeeds
- [x] User chooses to share their new "Username" & "Country" from the Layer8 Resource Server
- [x] Clicking "Get Next Poem" loads different poem correctly x 3
- [x] Clicking "Logout" takes you to the login page
- [x] Log in with 'tester' / '1234', and then with the credentials from Section 1 succeeds
- [x] This time, DO NOT share "display name" or "country"
- [x] Clicking "Get Next Poem" loads different poem correctly x 3

### SECTION 3: IMSHARER (http://localhost:5174/)
- [x] Main page loads
- [x] Upload of image works
- [x] Reload leads to instant reload (demonstrating proper caching)
- [x] Clicking the newly loaded image shows it in a lightbox

### Feature: Complete User "Happy Path" Flow Through Layer8, WGP, & Imsharer in Cypress

**Issue**: The user "Happy Path" flow through Layer8, WGP, and Imsharer needs to be automated using Cypress to ensure seamless user experience and functionality.

**Changes Made**:
1. **Layer8**:
   - Added Cypress tests for user login and initial navigation within Layer8.
   - Verified that users can successfully access and interact with key features.
   
**Related Issues**:
- #16

### CODE REVIEW

File: .github/workflows/e2e-test.yml [hunk 1]
```diff
@@ -94,3 +94,37 @@ jobs:
...
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '20'
+
+      - name: Install dependencies
+        run: npm install
+        working-directory: ./server/assets-v1/templates
+      
+      - name: Install cypress
```
Comment (stravid87) [inline .github/workflows/e2e-test.yml:113]: I notice that you "Install Cypress". Daniel, "usues, 
```
- name: Run cypress tests
        uses: cypress-io/github-action@v6
```
Am I missing something? You also have this same action later.... Why doesn't he also install cypress?

File: test/cypress/cypress.config.js [hunk 2]
```diff
@@ -1,9 +1,57 @@
...
+    devServer: {
+      framework: "vue-cli",
+      bundler: "webpack",
+    },
+  },
+
   e2e: {
     setupNodeEvents(on, config) {
-      // implement node event listeners here
+      on("task", {
```
Comment (stravid87) [inline test/cypress/cypress.config.js:17]: Love this. These are the tricks that can take hours to learn and there is no guarantee you'll find "this" solution when hustling to make something work.

File: .github/workflows/e2e-test.yml (hunk 1 above)
Comment (DeeStarks) [reply .github/workflows/e2e-test.yml:113]: Yes, the cypress image should handle the run

File: server/resource_server/service/service.go [hunk 3]
```diff
@@ -165,7 +165,7 @@ func (s *service) ProfileClient(userName string) (models.ClientResponseOutput, e
 	clientModel := models.ClientResponseOutput{
 		ID:          clientData.ID,
 		Secret:      clientData.Secret,
-		Name:        clientData.Username,
+		Name:        clientData.Name,
```
Comment (DeeStarks) [inline server/resource_server/service/service.go:168]: Aren't there any implications with this? Maybe parts of the server that currently expect username from here?

**Review by DeeStarks — COMMENTED** [review]
Looks good to me! Great work!

File: server/resource_server/service/service.go (hunk 3 above)
Comment (NJavokhir) [reply server/resource_server/service/service.go:168]: Frontend is waiting Name, that's why I changed Username to Name, I checked and it is working now.

**Review by DeeStarks — APPROVED** [review]

**Pull request merged**